tripy/examples/segment-anything-model-v2/sam2/build_sam.py

```diff
@@ -0,0 +1,462 @@
+        if not comp_info["enabled"]:
+            continue
+
+        executable_file = os.path.join(saved_engines_path, comp_name)
+        if os.path.exists(executable_file):
+            print(f"Loading existing compiled {comp_name} from {executable_file}")
+            compiled_model = tp.Executable.load(executable_file)
+        else:
+            print(f"Compiling {comp_name}...")
+            start = time.time()
+            compiled_model = tp.compile(comp_info["model"], args=comp_info["compile_args"])
+            print(f"Compilation took {time.time() - start:.2f}s")
+            compiled_model.save(executable_file)
+
+        old_model = comp_info["model"]
+        # If model is model.forward, retrieve the original model object
+        if hasattr(old_model, "__self__"):
+            old_model = old_model.__self__
+
+        set_model_attr(model, comp_name, compiled_model)
+        if "special_handling" in comp_info and comp_info["special_handling"] is not None:
+            comp_info["special_handling"](old_model)
+
+    model = model.to(device)
+    if mode == "eval":
+        model.eval()
+    return model
+
+
+def load_component_weights(comp_name, component_info, state_dict, checkpoint_dict):
+    """
+    Load weights for a single component from checkpoint into state dict.
+    """
+
+    converted_keys = 0
+    for key in checkpoint_dict:
+        # Remove _true/_false suffixes if present
+        for suffix in ["_true", "_false", ".compiled_executable"]:
+            if comp_name.endswith(suffix):
+                comp_name = comp_name[: -len(suffix)]
+                break
+
+        if not key.startswith(comp_name) or (component_info.get("skip_load_state_dict") is True):
+            continue
+
+        new_key = key.replace(f"{comp_name}.", "")
+        if "special_key_loading" in component_info:
+            new_key = component_info["special_key_loading"](new_key)
+        weight = checkpoint_dict[key]
+
+        should_convert = not any(skip in key for skip in component_info["skip_dtype_convert"])
+        if should_convert and component_info["dtype"] is not None:
+            weight = weight.to(getattr(torch, component_info["dtype"]))
+
+        state_dict[new_key] = tp.Parameter(weight.contiguous())
+        converted_keys += 1
+
+    return converted_keys
+
+
+def _load_checkpoint(model, ckpt_path, cfg=None):
+
+    if ckpt_path is None:
+        return
+
+    sd = torch.load(ckpt_path, map_location="cpu")["model"]
+    missing_keys, unexpected_keys = model.load_state_dict(sd, strict=False)
+
+    # Get paths for compiled models
+    current_dir = os.getcwd()
+    saved_engines_path = os.path.join(current_dir, "saved_engines")
+
+    # Get component configurations
+    components = get_component_configs(model, cfg)
+
+    # Process each component
+    for comp_name, comp_info in components.items():
+
+        if not comp_info["enabled"] or comp_info.get("skip_load_state_dict") is True:
```

Comment:
`is True` is probably not needed here:
```suggestion
        if not comp_info["enabled"] or comp_info.get("skip_load_state_dict"):
```